indico/modules/events/registration/controllers/management/fields.py

```diff
@@ -51,6 +52,11 @@ def _check_retention_period(self, retention_period, **kwargs):
             if field.type == RegistrationFormItemType.field_pd and field.personal_data_type.is_required:
                 raise ValidationError('Cannot add retention period to required field')
 
+            regform_retention_period = field.registration_form.retention_period
+            if retention_period > regform_retention_period:
+                raise ValidationError('Retention period cannot be longer than that of registration form',
```

Comment:
this one needs i18n since i don't think we validate it on the client side (which if fine IMO, we don't have access to the global retention period there)